src/App.test.tsx

```diff
@@ -1,9 +1,89 @@
-import React from 'react';
 import { render, screen } from '@testing-library/react';
 import App from './App';
+import userEvent from '@testing-library/user-event';
+import configureMockStore, { MockStore } from "redux-mock-store";
+import { Provider } from 'react-redux';
+import { authUser } from './store/userSlice';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+const mockUseNavigate = jest.fn();
+const mockStore = configureMockStore();
+let initialState = {};
+let store: MockStore;
+
+jest.mock('react-router-dom', () => ({
+  useNavigate: () => mockUseNavigate,
+  __esModule__: true
+}))
+
+const setup = (initialState = {}) => {
+  store = mockStore(initialState) as MockStore;
```

Comment:
If I change this, how can I use getActions() on the store?
Wouldn't initializing the store every time be redundant?

Reply:
You do it anyway (this is what `store = mockStore(...)` does, it creates a new mock store), you just also change a variable now (in addition to returning it (`return { store }` a few lines below)
Since you return it anyway, just use the returned value to get the store in the test and call its `getActions()` as usual. And then you do not need `let store;`

So in the test, you can use it this way -
```
it('...', () => {
 const {store} =  setup();
 expect(store.getActions()).to...
);
```